client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-name.js

```diff
@@ -52,72 +51,47 @@ export const getZoneName = ( zone, locations, translate, returnEmpty = false ) =
 	return locationNames.join( ', ' );
 };
 
-class ShippingZoneName extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.state = {
-			editing: false,
-		};
-	}
-
-	render() {
-		const { loaded, isRestOfTheWorld, zone, locations, actions, translate } = this.props;
-		const { editing } = this.state;
-
-		const startEditing = () => ( this.setState( { editing: true } ) );
-		const stopEditing = () => ( this.setState( { editing: false } ) );
-		const onNameChange = ( event ) => {
-			actions.changeShippingZoneName( event.target.value );
-		};
-
-		const renderContent = () => {
-			if ( ! loaded ) {
-				return (
-					<div className="shipping-zone__name is-placeholder">
-						<span />
-						<Button borderless>
-							<Gridicon icon="pencil" size={ 24 } />
-						</Button>
-					</div>
-				);
-			}
-
-			if ( editing ) {
-				return (
-					<div className="shipping-zone__name">
-						<FormTextInput
-							value={ getZoneName( zone, locations, translate, true ) }
-							onChange={ onNameChange }
-							placeholder={ translate( 'Enter a new zone name or leave empty to automatically list locations' ) } />
-						<Button borderless onClick={ stopEditing }>
-							<Gridicon icon="checkmark" size={ 24 } />
-						</Button>
-					</div>
-				);
-			}
+const ShippingZoneName = ( { loaded, zone, locations, actions, translate } ) => {
+	const onNameChange = ( event ) => {
+		actions.changeShippingZoneName( event.target.value );
+	};
 
+	const renderContent = () => {
+		if ( ! loaded ) {
 			return (
-				<div className="shipping-zone__name">
-					<span>{ getZoneName( zone, locations, translate ) }</span>
-					{ isRestOfTheWorld
-						? null
-						: <Button borderless onClick={ startEditing }><Gridicon icon="pencil" size={ 24 } /></Button> }
+				<div className="shipping-zone__name is-placeholder">
+					<span />
 				</div>
 			);
-		};
+		}
+
+		const zoneName = getZoneName( zone, locations, translate, true );
 
 		return (
+			<div className="shipping-zone__name">
+				<FormTextInput
+					value={ zoneName }
+					onChange={ onNameChange }
+					placeholder={ ! zoneName && getZoneName( zone, locations, translate ) } />
```

Comment:
The `! zoneName` condition is not needed, you can just always set the `placeholder` prop, it will only be shown when there's no real text written.